### PR TITLE
4 setup pylint

### LINKED ISSUE #4: setup pylint
Setting up pylint

### PR DESCRIPTION
**Pull request by Guthax**

### CODE REVIEW

File: src/models/train.py [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
 from joblib import dump, load
 
 def main():
+    """
+    Gets the model and trains it given the training and validation data.
+    """
 
     # Get model
-    model, params = getModel()
+    model = get_model()
```
Comment (SagaRut) [inline src/models/train.py]: We need the params

File: src/models/train.py [hunk 2]
```diff
@@ -1,10 +1,15 @@
-from model_definition import getModel
+"""Module that trains model given training data."""
+from model_definition import get_model
+from model_params import params
```
Comment (SagaRut) [inline src/models/train.py]: This is old, the file doesnt exist anymore so the params are from get_model()

File: src/models/model_definition.py [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+    Returns:
+        Sequential: Defined sequential model
+    """
+def get_model():
+    """
+    Builds the model from scratch using the keras library.
+
+    Returns:
+        Sequential: Defined sequential model
+    """
```
Comment (SagaRut) [inline src/models/model_definition.py]: why is this duplicated?

File: src/data/get_data.py [hunk 4]
```diff
@@ -1,24 +1,34 @@
+"""Module retrieving the data and persisting it to single files."""
+"""Module retrieving the data and persisting it to single files."""
```
Comment (SagaRut) [inline src/data/get_data.py]: Why is this duplicated?

File: report.txt [hunk 5]
```diff
@@ -0,0 +1,83 @@
+
+
+Report
+======
+117 statements analysed.
+
+Statistics by type
+------------------
```
Comment (SagaRut) [inline report.txt]: I see that this file is in the gitignore but you are still merging this to main? Also is this automatically generated with dslint? Also on mattermost, luis said that we could add these findings to the readme

File: .pylintrc [hunk 6]
```diff
@@ -0,0 +1,511 @@
+[MASTER]
```
Comment (SagaRut) [inline .pylintrc:1]: Did you change this at all from the original? bc we need to propose new ML rules. Also the decisions need to be properly motivated somewhere. For now just in the ReadMe. Also need to implement diff ways to display code quality info, maybe the report.txt is enough tho idk. This is from the assignment:

 The project implements different ways to display code quality information, considers multiple linters, critically analyses linter rules, and proposes new missing ML rules

File: .pylintrc [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+
```
Comment (SagaRut) [inline .pylintrc:2]: Also add instruction how to run pylint to the ReadMe

**Review by SagaRut — DISMISSED** [review]

File: report.txt [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+Duplication
+-----------
+
++-------------------------+------+---------+-----------+
+|                         |now   |previous |difference |
++=========================+======+=========+===========+
+|nb duplicated lines      |0     |0        |0          |
++-------------------------+------+---------+-----------+
+|percent duplicated lines |0.000 |0.000    |=          |
```
Comment (blibliboe) [inline report.txt]: How come there are no duplicates line according to the report, but there are in the actual files

**Comment by Timdnb** [comment]
If you update the things that @SagaRut pointed out, then it should be good, make sure to properly resolve the conflicts when merging tho

File: .pylintrc (hunk 6 above)
Comment (Guthax) [reply .pylintrc:1]: I used rules from https://github.com/SERG-Delft/dslinter/tree/main/docs/pylint-configuration-examples/pylintrc-for-ml-projects which is a pylintrc configured for machine learning projects.

File: report.txt (hunk 8 above)
Comment (Guthax) [reply report.txt]: it should be fixed now, it only analysis packages with an __init__.py and now they are added so it should analyse everything. Can you double check if this works for you now?

File: report.txt (hunk 5 above)
Comment (Guthax) [reply report.txt]: I deleted it, we can add it at the last moment when all other changes are done

File: report.txt (hunk 8 above)
Comment (blibliboe) [reply report.txt]: It seems to work fine for me now

**Review by blibliboe — APPROVED** [review]
Lgtm now

**Pull request merged**